src/Fantomas/AstTransformer.fs

```diff
@@ -62,1634 +34,1211 @@ module private Ast =
+                      yield! (List.collect id nodes) ]
+                    |> finalContinuation
+
+                Continuation.sequence continuations finalContinuation
+            | SynModuleDecl.Let (_, bindings, range) ->
+                mkNode SynModuleDecl_Let range
+                :: (bindings |> List.collect visitSynBinding)
+                |> finalContinuation
+            | SynModuleDecl.DoExpr (_, expr, range) ->
+                mkNode SynModuleDecl_DoExpr range
+                :: visitSynExpr expr
+                |> finalContinuation
+            | SynModuleDecl.Types (typeDefs, range) ->
+                mkNode SynModuleDecl_Types range
+                :: (typeDefs |> List.collect visitSynTypeDefn)
+                |> finalContinuation
+            | SynModuleDecl.Exception (exceptionDef, range) ->
+                mkNode SynModuleDecl_Exception range
+                :: (visitSynExceptionDefn exceptionDef)
+                |> finalContinuation
+            | SynModuleDecl.Open (target, parentRange) ->
+                // we use the parent ranges here to match up with the trivia parsed
+                match target with
+                | SynOpenDeclTarget.ModuleOrNamespace (_, _range) ->
+                    mkNode SynModuleDecl_Open parentRange
+                    |> List.singleton
+                    |> finalContinuation
+                | SynOpenDeclTarget.Type (synType, _range) ->
+                    mkNode SynModuleDecl_OpenType parentRange
+                    :: (visitSynType synType)
+                    |> finalContinuation
+            | SynModuleDecl.Attributes (attrs, range) ->
+                mkNode SynModuleDecl_Attributes range
+                :: (visitSynAttributeLists range attrs)
+                |> finalContinuation
+            | SynModuleDecl.HashDirective (hash, range) ->
+                [ mkNode SynModuleDecl_HashDirective range
+                  visitParsedHashDirective hash ]
+                |> finalContinuation
+            | SynModuleDecl.NamespaceFragment (moduleOrNamespace) ->
+                visitSynModuleOrNamespace moduleOrNamespace
+                |> finalContinuation
+
+        visit ast id
+
+    and visitSynExpr (synExpr: SynExpr) : TriviaNodeAssigner list =
+        let rec visit
+            (synExpr: SynExpr)
+            (finalContinuation: TriviaNodeAssigner list -> TriviaNodeAssigner list)
+            : TriviaNodeAssigner list =
+            match synExpr with
+            | SynExpr.Paren (expr, _, _, range) ->
+                visit
+                    expr
+                    (fun nodes ->
+                        [ mkNode SynExpr_Paren range
+                          yield! nodes ]
+                        |> finalContinuation)
+            | SynExpr.Quote (operator, _, quotedSynExpr, _, range) ->
+                let continuations : ((TriviaNodeAssigner list -> TriviaNodeAssigner list) -> TriviaNodeAssigner list) list =
```

Comment:
Does it makes sense to use the sequence helper function when the list is fixed and small (2 elements in this case)?